src/controller/quota/controller.go

```diff
@@ -167,107 +164,46 @@ func (c *controller) List(ctx context.Context, query *q.Query, options ...Option
 	return quotas, nil
 }
 
-func (c *controller) getReservedResources(ctx context.Context, reference, referenceID string) (types.ResourceList, error) {
-	conn := redislib.DefaultPool().Get()
-	defer conn.Close()
-
-	key := reservedResourcesKey(reference, referenceID)
-
-	str, err := redis.String(conn.Do("GET", key))
-	if err == redis.ErrNil {
-		return nil, nil
-	} else if err != nil {
-		return nil, err
-	}
-
-	return types.NewResourceList(str)
-}
-
-func (c *controller) setReservedResources(ctx context.Context, reference, referenceID string, resources types.ResourceList) error {
-	conn := redislib.DefaultPool().Get()
-	defer conn.Close()
-
-	key := reservedResourcesKey(reference, referenceID)
-
-	reply, err := redis.String(conn.Do("SET", key, resources.String(), "EX", int64(c.reservedExpiration/time.Second)))
-	if err != nil {
-		return err
-	}
-
-	if reply != "OK" {
-		return fmt.Errorf("bad reply value")
-	}
-
-	return nil
-}
-
-func (c *controller) reserveResources(ctx context.Context, reference, referenceID string, resources types.ResourceList) error {
-	reserve := func(ctx context.Context) error {
-		q, err := c.quotaMgr.GetByRefForUpdate(ctx, reference, referenceID)
+func (c *controller) updateUsage(ctx context.Context, reference, referenceID string, op func(hardLimits, used types.ResourceList) (types.ResourceList, error)) error {
```

Comment:
name it as `retryUpdateUsage` or `updateUsageWithRetry`? Then the caller could know this func with retry.